utils/wasm-gen/src/generator/syscalls.rs

```diff
@@ -82,9 +87,85 @@ impl InvocableSysCall {
                     ParamType::Delay,        // Number of blocks to delay the sending for
                     ParamType::Gas,          // Amount of gas to reserve
                     ParamType::Duration,     // Duration of the reservation
+                    ParamType::Ptr(None),    // Address of error returned
                 ]),
                 _ => unimplemented!(),
             },
         }
     }
+
+    fn into_info(self) -> SysCallInfo {
+        let underlying_syscall = match self {
+            Self::Loose(sc) => sc,
+            Self::Precise(sc) => sc,
+        };
+
+        let fallible = is_syscall_fallible(underlying_syscall);
+        SysCallInfo {
+            signature: self.into_signature(),
+            fallible,
+        }
+    }
+}
+
+// If syscall changes from fallible into infallible or vice versa in future,
+// we'll see it by analyzing code coverage stats produced by fuzzer.
+fn is_syscall_fallible(syscall: SysCallName) -> bool {
```

Comment:
Maybe we can introduce a new `SysCallName` method?

Reply:
We don't check anywhere in this `fn` or in tests that these syscalls are divided into fallible/infallible correctly. On the level of `runtime-fuzzer` we will see by analyzing coverage if there's errors, but if I place it on the level of `SysCallName` (inside `wasm-instrument`), then anyone can use this unchecked `fn` freely

Reply:
We can do it in another PR and write tests for `SysCallName::is_fallible()`.

Reply:
But for now fallible/infallible syscall division makes sense only for `wasm-gen`, and asserting this is not a trivial task. So I think we can have this solution